src/DeploymentSafetyEnforcer.function.ts

```diff
@@ -550,8 +550,9 @@ const execute = async (
       requestId,
       pipelineName: settings.pipelineName,
       rejectedBakeActions: rejectBakeActions.length,
-      settings: settings.metricsSettings ?? {
+      settings: {
         namespace: 'DeploymentSafetyEnforcer',
+        ...settings.metricsSettings ?? {},
```

Comment:
I don't believe you need the `?? {}`, since the spread operator will do nothing if `settings.metricSettings` is undefined